From 663c514fdc46a73b726d67851ee405557b37b5d0 Mon Sep 17 00:00:00 2001
From: Julian Scheel <julian@jusst.de>
Date: Wed, 4 Apr 2012 10:33:39 +0200
Subject: [PATCH 1/5] Fix possible double free.

currentit was freed but not NULLed, so it could be freed twice, which leads to
a crash.
---
 gst/mpegtsdemux/tsdemux.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/gst/mpegtsdemux/tsdemux.c b/gst/mpegtsdemux/tsdemux.c
index 96eb75c..3855d03 100644
--- a/gst/mpegtsdemux/tsdemux.c
+++ b/gst/mpegtsdemux/tsdemux.c
@@ -1524,6 +1524,7 @@ gst_ts_demux_push_pending_data (GstTSDemux * demux, TSDemuxStream * stream)
     g_list_free (stream->currentlist);
     gst_buffer_list_iterator_free (stream->currentit);
     gst_buffer_list_unref (stream->current);
+    stream->currentit = NULL;
     goto beach;
   }
 
@@ -1536,6 +1537,7 @@ gst_ts_demux_push_pending_data (GstTSDemux * demux, TSDemuxStream * stream)
   gst_buffer_list_iterator_add_list (stream->currentit, stream->currentlist);
   stream->currentlist = NULL;
   gst_buffer_list_iterator_free (stream->currentit);
+  stream->currentit = NULL;
 
   firstbuffer = gst_buffer_list_get (stream->current, 0, 0);
 
-- 
1.7.10

